src/can/can.c

```diff
@@ -88,20 +128,34 @@ void init_can() {
         CANSTMOB = 0x00;
     }
 
+    // set global interrupt flag
     sei();
 
+    // set can to enable mode
     CANGCON |= _BV(ENASTB);
-    while (!(CANGSTA & _BV(ENFG))) {}
-    // enable CAN and wait for CAN to turn on before returning
+
+    uint16_t timeout = 1;
+
+    while (!(CANGSTA & _BV(ENFG)) & !(timeout == 0)) {
+        // enable CAN and wait for CAN to turn on before returning
+        // added a timeout for this infinite loop
+    }
```

Comment:
This doesn't work as a timeout, because `timeout` is a variable that you need to increment/decrement yourself. See https://utat-ss.readthedocs.io/en/master/style-guide/technical.html#infinite-loops-and-timeouts as an example.